coderd/coderdtest/coderdtest.go

```diff
@@ -570,6 +572,36 @@ func NewWithAPI(t testing.TB, options *Options) (*codersdk.Client, io.Closer, *c
 	return client, provisionerCloser, coderAPI
 }
 
+// AwaitProvisionerDaemonsConnected waits for the provisioner daemon to connect.
+func AwaitProvisionerDaemonsConnected(t testing.TB, api *coderd.API) {
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+	defer cancel()
+	for {
```

Comment:
Instead of this polling loop with potentially noisy logs, what if we made a slight addition to the `provisionerd.Server` such that it closes or sends on a buffered channel when it connects for the first time. This is how the tailnet dialer works.

 Then we could select on that channel (vs `ctx.Done()`) rather than polling.

Reply:
I tried this out, except at the provisionerd level itself. LMK if that suits, or I can circle back and do it in the provisionerd server as you suggested.